src/Analyzers/CSharp/Analyzers/NamingStyle/CSharpNamingStyleDiagnosticAnalyzer.cs

```diff
@@ -20,5 +25,14 @@ internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosti
                 SyntaxKind.LocalFunctionStatement,
                 SyntaxKind.Parameter,
                 SyntaxKind.TypeParameter);
+
+        // Parameters of positional record declarations should be ignored because they also
+        // considered properties, and that naming style makes more sense
+        protected override bool ShouldIgnore(ISymbol symbol)
+            => symbol.IsKind(SymbolKind.Parameter)
+            && IsParameterOfRecordDeclaration(symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax());
+
+        private static bool IsParameterOfRecordDeclaration(SyntaxNode? node)
+            => node != null && node.HasAncestor<RecordDeclarationSyntax>();
```

Comment:
❗  This will end up excluding all parameters at any level of a record. The filter should be more specific.

Reply:
Good point, I completely missed that.